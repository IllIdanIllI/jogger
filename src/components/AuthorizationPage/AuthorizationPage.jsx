import React from 'react';
import './AuthorizationPage.scss';
import bearFace from '../../images/bear-face/bear-face.png';

const AuthorizationPage = () => {
    return (
        <div>
            <div className="login-form-wrapper">
                <img src={bearFace} className="login-form-wrapper__img" />
                <button className="login-form-wrapper__btn">
                    Let me in
                </button>
            </div>
        </div>
    );
};

export default AuthorizationPage;
