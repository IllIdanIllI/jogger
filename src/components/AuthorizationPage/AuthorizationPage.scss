$baby-purple: #e990f9;
$white-two: #ffffff;
.login-form-wrapper {
    display: grid;
    width: 503px;
    height: 379px;
    border-radius: 44px;
    background-color: $baby-purple;
    justify-items: center;
    align-items: center;
    transform: translate(100%, 50%);
}
.login-form-wrapper__img {
    width: 160px;
    height: 150px;
    object-fit: contain;
}

.login-form-wrapper__btn {
    width: 151px;
    height: 60px;
    border-radius: 36px;
    border: solid 3px $white-two;
    background-color: inherit;
    color: $white-two;
    font-size: 18px;
    font-weight: bold;
    cursor: pointer;
}
