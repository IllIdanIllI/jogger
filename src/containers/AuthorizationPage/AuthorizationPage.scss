@import '../../style/colors';
@import '../../style/commonVars';

div {
    position: relative;
    .login-form-wrapper {
        position: fixed;
        display: grid;
        width: 503px;
        height: 379px;
        border-radius: 44px;
        background-color: $baby-purple;
        justify-items: center;
        align-items: center;
        left: 50%;
        top: 50%;
        transform: translate(-50%, -50%);

        &__img {
            width: 160px;
            height: 150px;
            object-fit: contain;
        }

        &__btn {
            width: 151px;
            height: 60px;
            border-radius: 36px;
            border: solid 3px $white-two;
            background-color: inherit;
            color: $white-two;
            font-size: 18px;
            font-weight: bold;
            cursor: pointer;
        }
    }

    @media (max-width: $mediaWidth) {
        div {
            height: 100%;
            .login-form-wrapper {
                position: inherit;
                grid-template-rows: min-content min-content;
                grid-row-gap: 10vh;
                width: 100%;
                height: 100%;
                top: unset;
                left: unset;
                transform: none;
                background-color: white;
                top: 100pt;
                align-items: baseline;
                &__img {
                    object-fit: contain;
                    g path {
                        fill: $baby-purple;
                    }
                }

                &__btn {
                    border-radius: 40px;
                    border: solid 3px $baby-purple;
                    background-color: inherit;
                    color: $baby-purple;
                    font-size: 24px;
                    font-weight: bold;
                    cursor: pointer;
                }
            }
        }
    }
}
