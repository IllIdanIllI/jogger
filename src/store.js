import { createStore, combineReducers, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import authentication from './reducers/authenticationReducer';
import jogStore from './reducers/jogsReducer';
const rootReducer = combineReducers({
    authentication,
    jogStore,
});

const middleware = [thunk];

const store = createStore(
    rootReducer,
    composeWithDevTools(applyMiddleware(...middleware))
);
export default store;
