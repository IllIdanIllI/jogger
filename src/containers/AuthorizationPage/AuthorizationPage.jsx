import React, { useEffect } from 'react';
import { Redirect } from 'react-router-dom';
import { useSelector } from 'react-redux';
import './AuthorizationPage.scss';
import { ReactComponent as BearFace} from '../../images/bear-face/bear-face.svg';
import { useActions } from '../../custom/customHooks';
import {
    authenticate,
    reAuthenticate,
} from '../../actions/authenticationActions';
import { TOKEN } from '../../constants/constants';

const AuthorizationPage = ({ isAuthenticate }) => {
    const [onAuthenticate, onReAuthenticate] = useActions([
        authenticate,
        reAuthenticate,
    ]);

    const isAuthenticated = useSelector(
        (state) => state.authentication.isAuthenticated
    );

    useEffect(() => {
        const token = localStorage.getItem(TOKEN);

        if (token && !isAuthenticate) {
            onReAuthenticate(token);
        }
    }, [isAuthenticated]);

    if (isAuthenticated) return <Redirect to="/jogs" />;

    return (
        <div>
            <div className="login-form-wrapper">
                <BearFace className="login-form-wrapper__img"/>
                {/* <img src={bearFace} className="login-form-wrapper__img" alt="Bear" /> */}
                <button
                    onClick={() => onAuthenticate()}
                    className="login-form-wrapper__btn"
                >
                    Let me in
                </button>
            </div>
        </div>
    );
};

export default AuthorizationPage;
