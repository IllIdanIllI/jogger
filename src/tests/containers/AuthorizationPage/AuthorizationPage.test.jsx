import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../../store';
import AuthorizationPage from '../../../containers/AuthorizationPage';
import renderer from 'react-test-renderer';

test('AuthorizationPage test', () => {
    const component = renderer.create(
        <Provider store={store}>
            <Router>
                <AuthorizationPage />
            </Router>
        </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
