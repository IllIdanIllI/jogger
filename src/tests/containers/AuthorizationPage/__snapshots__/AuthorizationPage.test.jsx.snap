// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorizationPage test 1`] = `
<div>
  <div
    className="login-form-wrapper"
  >
    <svg
      className="login-form-wrapper__img"
    >
      bear-face.svg
    </svg>
    <button
      className="login-form-wrapper__btn"
      onClick={[Function]}
    >
      Let me in
    </button>
  </div>
</div>
`;

exports[`Header test 1`] = `
<header>
  <a
    href="/"
    onClick={[Function]}
  >
    <svg
      className="header__logo "
    >
      logo.svg
    </svg>
  </a>
</header>
`;
